### ISSUE TITLE
No data? 

### ISSUE THREAD

**Issue by discodancerstu** [barleybobs/homeassistant-ecowater-softener#54]
Hi all, anyone else having issues with the integration? No data coming through into HA for last couple of days.

Portal appears to be working fine?!

**Comment by misterymisery** [comment]
It stopped to work for me yesterday. I can see salt quantity etc but in Energy tab it is not display Ing anything.

**Comment by barleybobs** [comment]
@discodancerstu Are you still not getting data? I have checked my instance and I am still receiving data.

**Comment by discodancerstu** [comment]
It is sporadic, some days its fine and other days not.  I find a reset of the water softener often helps, almost like it is losing WiFi - however it is less than 2 metres from the AP!

**Comment by barleybobs** [comment]
That is odd. Is there anything in HA logs?

**Comment by discodancerstu** [comment]
Sorry, where would I find that data?

**Comment by barleybobs** [comment]
It should be under Settings > System > Logs

**Comment by misterymisery** [comment]
Again no data in Energy tab from two days (I can see pattern 6 days works and stops). Nothing in logs and integration works and showing data. Strange thing...

**Comment by barleybobs** [comment]
I've not be having a similar issue so I am not sure what could be causing this. It's odd that no logs are being produced.

**Comment by ijosbuttler** [comment]
If you're experiencing issues with the integration not pulling data into Home Assistant, there could be several potential reasons for this problem:

1. Integration Configuration: Double-check that the integration is correctly configured in Home Assistant. Ensure that the login information and settings are accurate, including the serial number and date format.
2. Network Connectivity: Verify that your Home Assistant instance has internet access and can connect to the Ecowater portal. If there are any network issues or firewall rules blocking the connection, it could prevent data from being retrieved.
3. Ecowater Portal Status: While the portal may appear to be working fine, there could be backend issues or maintenance activities affecting data retrieval. Check ([watersoftenersizecalculator](https://watersoftenersizecalculator.com/)) if there are any announcements or service status updates from Ecowater regarding potential disruptions.
4. Integration Updates: Ensure that you're using the latest version of the Ecowater integration. Sometimes updates or changes on the service provider's side may require corresponding updates to the integration for seamless operation.
5. Logs and Error Messages: Check the Home Assistant logs for any error messages related to the Ecowater integration. This can provide valuable insight into what might be causing the issue.

If none of the above steps resolve the issue, you may need to troubleshoot further.

**Comment by ozgersefa** [comment]
Keeping track of your Emirates ID status has never been easier, thanks to the UAE government's convenient online system. Gone are the days of long queues and complicated processes – now you can monitor your application from the comfort of your own home, saving both time and hassle. No need to fuss over special registrations or passwords; it's all straightforward.

To check your Emirates ID card status, follow these simple steps:

Visit [https://emiratesidstatuscheck.ae/] or official website of the UAE government dedicated to Emirates ID services.

**Comment by dragosj** [comment]
I also have same issue, no data since two days ago and in mobile app or web everything is fine. Energy dashboard does not show anything anymore.
I get a warning: Setup of sensor platform ecowater_softener is taking over 10 seconds.
HA does not show any errors on all 9 entities discovered in evowater.

**Comment by barleybobs** [comment]
There appears to be an issue on EcoWater's end. As I have just checked and the web interface is claiming that 0 litres of water have been used but the graph shows the correct value. Therefore I believe this is an issue not with the integration but with EcoWater's website.

**Comment by dragosj** [comment]
Hi, I can confirm I think it was something from their end. I unpluged for 1 min. the appliance and then it start sending data. 2 days are lost, no reports.

**Comment by BelMaseto** [comment]
Same here since four days: 0 daily liters  used but graph values are OK.

**Comment by HumanSkunk** [comment]
Same issue here. Thought it was the integration but it’s definitely something server side. I have had to restart the softener every 36 hours or so over the last week as data seems to start going wrong. The softener reports the right information on device and the graphs show the correct usage but the the current water used and the average will just show some random number that it has got stuck on, usually 0L. A power cycle instantly fixes it. In the last restart I went through the wireless setup again to see if that kicks anything into life.

frustrating as I had this installed only a month ago and one of the features I wanted was usage data etc. At least I know others have the same problem and it isn’t isolated to my machine. At least  it still functions as a softener…

**Comment by Ekkib** [comment]
I also can confirm : no data anymore. Since approx. 22 May. A powercycle of the softener triggers data again for some very short time. No idea, what goes wrong ?

**Comment by HumanSkunk** [comment]
I am tempted to temporarily disable the integration if it does it again and see if there is the same behaviour. I can confirm it stopped working reliably at 12am on 23rd May, it stopped again at 5am on 29th May, and then at 12am today.

**Comment by BelMaseto** [comment]
More on this issue.

Looking into the device's EASE report available on my EcoWatter account I've seen that at some point, the  Controller Time data listed there is stuck to a fixed time value (in my case was about 03:10) and when this happens, daily consumption value is stuck too.

On normal condition Controller Time field shows current time value, so seems that the issue is related to a bug in the device firmware that locks the device clock.

Power cycle the device solves this. As a workaround I will place a Shelly there to reset the softener daily.

**Comment by HumanSkunk** [comment]
Failed again last night. So do they remotely update as it was fine for 3.5 weeks after the install? I have a spare smart plug so I can fit that. Has anyone reported this issue to EcoWater? My contact all goes direct through the installer.

**Comment by figorr** [comment]
> Failed again last night. So do they remotely update as it was fine for 3.5 weeks after the install? I have a spare smart plug so I can fit that.
> 
> On the web portal I also checked the device time and it’s currently reporting as 23:19 listed under support -> troubleshooting.
> 
> Has anyone reported this issue to EcoWater? My contact all goes direct through the installer.

A also saw this a week ago ... but after doing a power cycle ... the device starting sending the correct data to web, app and api ... so I supposed It was a problem with my network (maybe the firewall blocked something and the device was stucked). But today I saw this issue again ... I was checking and no firewall rules were applied.

I sent a request to the Ecowater Team using the Support Form you can find when you log at wifi.ecowater.com. The contact method from the app ... is not working (the app closes when you tap the button, iOS app).

**Comment by cedricdelecole** [comment]
Same here since 28/05 ... and indeed the ecowater app only shows "0 liters used" days after days, but when you look at the graphs, it shows correct daily / hourly consumption.
One thing though is that I am running out of salt at 0%, but I assume that has no impact on the fact that 0 liters are showned as daily usage ...

![Screenshot_20240605_122219_EcoWater Systems (1)](https://github.com/barleybobs/homeassistant-ecowater-softener/assets/135960043/9f483ddd-5e81-4192-b8f9-a3e6a8895217)

![Screenshot_20240605_122230_EcoWater Systems (1)](https://github.com/barleybobs/homeassistant-ecowater-softener/assets/135960043/4b61b45d-7eac-4b0a-a734-0add3e7beca0)

I have tried contacting them via the app to report the issue but no feedback from them yet.
Have you done so ?

**Comment by barleybobs** [comment]
> P.S. I just contacted them at the number on this page (https://www.ecowater.be/fr/contact) to report the issue and they just solved it over the phone by asking me to unplug the power for 15 seconds then plug it back. The data came straight into the app back ...

@Ekkib mentioned that a power cycle appears to get data again for some very short time. It seems to be that a power cycle will fix the issue with data not arriving temporarily although then the data will stop later.

**Comment by BelMaseto** [comment]
It happened again after 4 days since last power cycle.
I can confirm that this issue is some way related to device internal Controller Time: EASE report shows that mine was stuck some hours ago, as it did every time daily consumption failed.
Power cycle again and solved...until next failure, so time to place a Shelly to force a daily reboot.

**Comment by almoney** [comment]
It seems like your controller is failing.  I don't have this issue.

**Comment by HumanSkunk** [comment]
I have also fitted a smart plug to mine which power cycles it every 24 hours at midnight. Hopefully until Ecowater acknowledge the issue beyond "restarting it" this should stop this issue. The installer has also raised it but he has said he hasn't seen it on any others that are being monitored.

**Comment by heisenberg2980** [comment]
I just installed an ecowater softener and got this issue. After adding the integration to HA the data was retrieved without any problem, but then it was stuck there and not updated for hours (even the data in the ecowater app was updated). Then I decided to disable and enable the integration, and that solved the issue, which means it was caused by the integration and not the ecowater servers. Could it be that we are dealing with multiple issues, one of them caused by ecowater, and a different one caused by the integration not retrieving the data properly?
Also is there a way to know when was the last time the integration retrieved data from the server? as it only happens every 10min, it would be useful to know when it´s happening to troubleshoot issues

**Comment by figorr** [comment]
> I just got an ecowater softener and this issue happened right after installing it. After adding the integration to HA the data was retrieved without any problem, but then it was stuck there and not updated for hours (while the data in the ecowater app was updated fine). Then I decided to disable and enable the integration, and that solved the issue, which means it was caused by the integration and not the ecowater servers. Could it be that we are dealing with multiple issues, one of them caused by ecowater, and a different one caused by the integration not retrieving the data properly?
> 
> Also is there a way to know when was the last time the integration retrieved data from the server? as it only happens every 10min, it would be useful to have an entity or similar showing when the last synchronization happened to troubleshoot issues (or even better, a button to allow a manual synchronization)

What you explained is a different issue. For instance ... today my softener was stucked again ... and the integration was not updating data but neither the iOS app was updating the data ... after swithing off and on the softener ... the data was inmediately updated (in HomeAssistant and in the iOS app) without reloading the integration.

Sometimes the Ecowater server goes down for a while, some minutes (you can see the device sensor is "Offline") ... and the next time the integration asked for data (10 minutes later) ... it works fine and gets the data without problem. If you reload the integration before the 10 min ... you can shorten this time and get the data again.

Regarding to the option ... to know when was the last time the integration retrieved data from the server ... could be solved creating a sensor, using yaml, that checks when an atributte, like "water used today" updates its value.

**Comment by heisenberg2980** [comment]
Regarding the sensor to know when the integration retrieved data for the server, I thought about that but it wouldn´t work. That would only give you a real value when there isn´t any issue with ecowater servers (so it wouldn´t help to troubleshoot issues) and if the water used today value changes, so it wouldn´t really be helpful. A real sensor that is updated every time the integration successfully retrieve data from the ecowater would be much more helpful.

**Comment by figorr** [comment]
Ok, I just sent a pull request in order to create a sensor for date and time data when data is updated from the server.. But I am bit new coding here ... so before merging I need someone else can check the changes. In order to validate that there is no a huge mess in that code. I will try to test the changes in my installation during the weekend, to see if there is nothing extrange.

**Comment by HumanSkunk** [comment]
So it seems doing a power cycle to restart at midnight is not enough as twice now the system clock has frozen during the afternoon and stops passing any usage data into the EcoWater servers.  Will increase the frequency to do it at midday as well to hopefully get more stable data.

**Comment by figorr** [comment]
Today they just released a new app release, at least at iOS. Let’s see if there are good news with server connection, and there is no more disconnections.

![IMG_9477](https://github.com/barleybobs/homeassistant-ecowater-softener/assets/77553975/b9237fab-8546-493f-b3c6-668c1cec4a04)

**Comment by HumanSkunk** [comment]
So having had the integration disabled since my last update I have had no issue with data feeds and the system clock remains at the correct time. I havent power cycled the softener since disabling the integration. Prior to this I couldn't get two days without data being interrupted. I will leave it another few days to get to close to 100% to no issue. I will then turn the integration back on and see what happens and report back.

**Comment by niallobr** [comment]
That iOS 1.25.1 update did not make any difference for me. I've now disabled the integration and will power cycle my unit too. Previously I did a power cycle and left the integration running. Data connection resumed for a while then stopped again

![Screenshot 2024-06-21 162248](https://github.com/barleybobs/homeassistant-ecowater-softener/assets/204252/0066ede0-04de-4dbc-9126-97943f08a4e1)


I would rather it is an issue with the integration as I recently had the Compact 200 softener fitted which was a lot more expensive than some of the alternatives and I was hoping for easy integration with HA, so hoping it is an simple integration fix rather than something complicated or unlikely to be fixed on the EcoWater side.

**Comment by laplac** [comment]
I have the same issue, I have reported it to portal support as well as ecowater customer service but still no response.

**Comment by HumanSkunk** [comment]
Since disabling the integration over a week ago the Softener has not needed to be restarted and has reported consistently over that time. I am now going to turn the integration back on and will report back if the softener freezes. I was wondering if it could be a memory leak issue, as I didnt encounter the first freeze for just under 3 weeks after the device was installed.

**Comment by figorr** [comment]
Today … since midday … no data. I tried several power cycles without success. I tried disabling the integration and doing another power cycle … but no difference.  The app also is showing the device as offline. But the softener is showing internet connection and the router is showing the device as connected. The weird thing … in the past when the update of the data failed … I can see, at the app, the water usage graphs being well updated according to the usage. Today, since midday, even the graphs are not being updated.

**Comment by niallobr** [comment]
My data is still working since I disabled the integration and restarted the softener. The data on the EcoWater app home screen doesn't always update right away. Sometimes I need to go to the graphs and back to home and it updates. Maybe they added some limits or security updates on the API? No idea, just speculating.

**Comment by laplac** [comment]
I have added a smart plug and scheduled an auto power-cycle everyday at 5:00 am. So far so good. 

For those who disabled the integration and enabled it again. Did it fix the issue ?

**Comment by heisenberg2980** [comment]
I think we are trying to solve a problem that is not in our control. I think ecowater has some type of issue in their software or servers that makes their servers stop receiving certain data from the water softener, and for some reason switching the device off and back on again solves the issue (I guess when the device starts, it sends some type of initial message to their servers that temporarily fix the issue). I say "certain data" because I have seen the ecowater app stuck on 0 liters, but at the same time the "graph" tab shows data about usage. The only solution that seems to work at the moment is the one @laplac mention about power-cycling the water softener everyday.

I am trying to contact ecowater to help them with this issue, and I will propose they update the firmware of the devices to include an option to configure a local MQTT server where the device can send real-time data without depending on their servers, which me luck!

**Comment by HumanSkunk** [comment]
I had it power cycling every night but still it would freeze during the day, then had it power cycling at lunch as well and it would still freeze between. Seems that the time between freezing just got quicker and quicker each time. I had the integration off for over a week with no clock freezing. Since turning the integration back on it’s been fine without power cycling), but it’s only been a couple of days. If and when it crashes I will report back. When it was first installed it took about 3 weeks for it to freeze the first time.

@heisenberg2980 good luck with contacting them. MQTT local integration would be great.

**Comment by HerrTim** [comment]
Hi,
I just opened a new issue but I then saw your conversation. I had a call with the german support this morning. They've told me, the sync freeze is caused by using a third party smart home system. That's what the us IT guys told him some days ago.

I removed the connection in home assistant but the problem still occurs in the ecowaters.com dashboard. The value of water consumption for today is still broken. The chart works but the value is not updating. It also shows an update error.
He will contact the us IT department again and keeps me in the loop.

**Comment by barleybobs** [comment]
> They've told me, the sync freeze is caused by using a third party smart home system. That's what the us IT guys told him some days ago. Ecowater officially does not accept any third-party tools.

I wonder if they've tried to block the integration. However, that sounds more like they won't provide any support for the integration.

**Comment by cedricdelecole** [comment]
This is something which always astonished me .... HA is making tons of polling to lots of brands and their servers, to get the latest entity values.
Cn you imagine the number of polls made every day by HA and the impact on the brand servers ????

Usually if I have an water softener like the Ecowater, I go to the app and connect few times a week (max ?).
How many polls is HA doing to the Ecowater servers to get the consumption .... ?

**Comment by HumanSkunk** [comment]
So the softener froze at 15:36 today. Only just noticed when I logged in remotely to check the status. Gonna have to disable it for the time being. I’m not particularly in needed of minute by minute consumption so perhaps an hourly update?

**Comment by barleybobs** [comment]
> This is something which always astonished me .... HA is making tons of polling to lots of brands and their servers, to get the latest entity values. Cn you imagine the number of polls made every day by HA and the impact on the brand servers ????
> 
> Usually if I have an water softener like the Ecowater, I go to the app and connect few times a week (max ?). How many polls is HA doing to the Ecowater servers to get the consumption .... ?

While I understand that HA polls the servers more than a human user. This would not be a problem if there was an official API or even better a local API.

**Comment by cedricdelecole** [comment]
by local API you mean we would poll the water softener directly without having to disturb the Ecowater servers ?
Didn't think about that

**Comment by barleybobs** [comment]
> by local API you mean we would poll the water softener directly without having to disturb the Ecowater servers ? Didn't think about that

Yes, a local API would allow us to talk to the device over the network directly. This would mean that Ecowater wouldn't even need to host an API in the cloud. It would also allow us to poll it as frequently as we liked.

**Comment by heisenberg2980** [comment]
There is even a better solution, and I am trying to contact ecowater to see if they would be open to implement that solution, as it would be amazing for Home Assistant user. This solution is for the device to have an option to send data to a MQTT broker. If they added this option, instead of having to pull the data by calling an API, the device will actually send the data to the local MQTT broker every time there is an update in any of the variables (like number of liters used, or changes in the water flow), so Home Assistant would receive the updates in real-time.

In a completely perfect world, they would implement both solutions (local API + connection to MQTT broker), as that would allow them to use the local API on their mobile app, so the app calls the local API when the user is home, reducing even more the load on their servers.

So far I have only been able to get attention from my local installer, but he told me a guy from ecowater Belgium would contact me. As I mentioned before, wish me luck...

**Comment by niallobr** [comment]
My integration has been disabled for around a week. Everything was going great but now the water usage on the EcoWater app and web dashboard is no longer updating. The graphs are still working fine. So in my mind it seems like the HA integration is _not_ causing this problem. Maybe EcoWater are making updates to their web platform in the background which is breaking things?

**Comment by misterymisery** [comment]
I have integration deleted completely for over a week and since than I have no issues at all. I sent email to support and they told me to delete HA integration and check. Also they told me that it is for hidden to use any 3rd party company apps or integration to EcoWater Softener. I the end I lost something which worked fine until May. I guess integration changed that time. I have to admit 30nminutes was just fine for me on beginning on integration with Python. Maybe these półka are too often or something... Anyway delete fixed it for me so I am no longer using integration... Sad :(

**Comment by BelMaseto** [comment]
The integration is based on ecowater-softener Python lib (https://pypi.org/project/ecowater-softener/), that just scraps data from the wifi.ecowater.com page like any browser does, so it's unlikely that this is causing this issue. 

IMO, this is caused by some poor/unfinished implementation at server end and/or device communication module itself.

**Comment by heisenberg2980** [comment]
I agree with @BelMaseto. I think we should stop trying to fix the issue ourselves, as it's 99% a server-side issue and the only one who can fix it is ecowater, and start trying to contact ecowater in every possible way we can imagine, so this request of local access to the data reach their management team. I am already putting all the pressure I can on my local installer and he is trying to get one of the ecowater guys to contact me, but maybe if you guys try the same thing through your local installers at least one of us might get lucky.

**Comment by BelMaseto** [comment]
> ... and start trying to contact ecowater in every possible way we can imagine, so this request of local access to the data reach their management team...

I would suggest to remark them that some customers  -myself- have mainly decided to purchased an Ecowater device because it can be integrated into HA thanks to barleybobs awesome work, as well as the fact that this integration that have cost them nothing gives their brand a very valuable competitive advantage.

**Comment by niallobr** [comment]
They seem to be difficult to contact. I sent an email to the UK/Ireland rep that is listed on the support page at wifi.ecowater.com and it bounced back. I've previously filled out the contact form and gotten no response back from them. Just filled it out again.

**Comment by HumanSkunk** [comment]
> > ... and start trying to contact ecowater in every possible way we can imagine, so this request of local access to the data reach their management team...
> 
> I would suggest to remark them that some customers -myself- have mainly decided to purchased an Ecowater device because it can be integrated into HA thanks to barleybobs awesome work, as well as the fact that this integration that hasn't cost them a penny gives their brand a very valuable competitive advantage.

This is also partly my justification for buying. I did a quick search on Softeners that could be integrated with HA, was happy with how it worked. I dont know why Ecowater are being so difficult about it, the only response I have had is to "turn it off for 15 mins and turn it back on again", honestly I am disappointed with their feedback thus far, they supply to consumers but seem to behave like they only deal with enterprise and will only be more forthcoming with the installer network than the end user. I have reached out to my installer but he doesn't really have any idea with what Home Assistant is or does but it is kinda my last ditch attempt.

As mentioned above if this integration works through web scraping, which I assume is being done by my local instance of HA then it seems bonkers that their systems would just pack up after x number of requests. We should be permitted to log in to the server as many times as we please... Perhaps that is why some peoples instances have failed even with this integration disabled, to many checks in a given period? Something has been badly programmed server side!

**Comment by misterymisery** [comment]
One thing is strange. When we have integration with 30 minutes interval it was fine. Now with smaller time Between reads it died for most of us. Maybe we could try with option in configuration for integration to choose between 10,30,60 minutes data gathering? We could set what works best?

**Comment by HumanSkunk** [comment]
Just an idea (I didnt realise @misterymisery had the same idea as well, can we lower the refresh time of the sensors? I looked through the code to see if there was somwhere that was obvious but couldn't spot it? Updating every 30/60 mins could be sufficient for just monitoring.

**Comment by ralphLe0n** [comment]
I changed the refresh time from 10 to 30 minutes, and the sensor worked longer than before, but in the end, it frozen. 
The part of the code that is responsible for the interval check is in coordinator.py line 33 

https://github.com/barleybobs/homeassistant-ecowater-softener/blob/main/custom_components/ecowater_softener/coordinator.py#L33

A workaround I'll be doing with this is creating the automation on the error log to power cycle the Softener.

**Comment by RobinLenaerts-r0627775** [comment]
Just to add some context here. I recently had contact with their support because I stumbled into this issue. They told me it was a security measure that is taken once the system detects it gets "hacked" (his words, not mine) when a third party connects to it. 
This explanation stinks of customer support BS to me but thought I might share it here

**Comment by ttuuxxeerr** [comment]
> I changed the refresh time from 10 to 30 minutes, and the sensor worked longer than before, but in the end, it frozen. The part of the code that is responsible for the interval check is in coordinator.py line 33
> 
> https://github.com/barleybobs/homeassistant-ecowater-softener/blob/main/custom_components/ecowater_softener/coordinator.py#L33
> 
> A workaround I'll be doing with this is creating the automation on the error log to power cycle the Softener.

I did the same and at least it works a few days more. thanks for sharing

**Comment by niallobr** [comment]
@ttuuxxeerr @ralphLe0n could you share your automation yaml please? Not sure how to check that it has gone silent, is it with Ping (ICMP)? Assuming you need a smart plug to power cycle the softener?

**Comment by ralphLe0n** [comment]
To be completely honest with you, I haven’t done it yet :’( but I was about to follow this thread on the community forum https://community.home-assistant.io/t/automation-based-on-log/337925/6.

**Comment by RobinLenaerts-r0627775** [comment]
@ttuuxxeerr @ralphLe0n @niallobr In case you still need it, I have a working automation now. been running for 1 day and it restarted the device yesterday after catching a log with the error message!

I followed @ralphLe0n s link on automations based on logs to enable that integration. After that I created the following:

My water softener was already plugged in through a smart plug but you will need one to make this work.
the device_id and entity_id point towards that smart plug.

Script 
```yaml
alias: Restart Water Softener
sequence:
  - type: turn_off
    device_id: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    entity_id: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    domain: switch
  - delay:
      hours: 0
      minutes: 0
      seconds: 10
      milliseconds: 0
  - type: turn_on
    device_id: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    entity_id: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
    domain: switch
description: ""
icon: mdi:water-opacity
```

Automation 
```yaml
alias: EcoWater auto-restart
description: ""
trigger:
  - platform: event
    event_type: system_log_event
condition:
  - condition: template
    value_template: >-
      {{ 'Error connecting to "wifi.ecowater.com"' in
      trigger.event.data.message[0] }}
action:
  - action: notify.mobile_app_robin_s_iphone
    metadata: {}
    data:
      message: Fuck their API restrictions
      title: Restarting water softener
  - action: script.restart_water_softener
    metadata: {}
    data: {}
mode: single
```

The notification is purely for my own enjoyment and can be ignored :)

Hope this helps someone!

**Comment by heisenberg2980** [comment]
@RobinLenaerts-r0627775 this is amazing, I will try it straight away, thanks!

I suggested in [this comment](https://github.com/barleybobs/homeassistant-ecowater-softener/issues/54#issuecomment-2165473190) for the integration to include a new sensor called "last update time" or similar, that we could use for automations, and it would be pretty simple "if last_updated_time more than 1h ago --> power cycle". @figorr you mentioned you created a pull request for this, what is the number of that PR?

**Comment by R1DEN** [comment]
Just to get some more info into the mix. I've contacted my dealer and after maaany back-and-forth messages I got the following: 

> It was corporately agreed that direct contact with any non-official apps is blocked

Unfortuntely I'm not in the US so have a very indirect path of contacting them.
Maybe we could somehow create a collective "petition" of some sort?
Even if there are 10k people using this integration (which there definitely aren't) it's an extremely small 16 requests per second on average, which shouldn't be an issue at all in 2024

**Comment by R1DEN** [comment]
@barleybobs or we can approach it from another end. the softener sends a request to ans-2-field.aylanetworks.com after which it requests xxx.yyy.device.aylanetworks.com (xxx.yyy may be an ID, so I've masked them).
It is possible to make a custom DNS rewrite (which will redirect to hassio server) + some trickery inside of nginx to detect the request and forward it to a specific port and run an addon there, that will forward the request later to the actual host (probably directly with an IP address because otherwise it will get caught right back by adguard).
this way we will intercept everything the softener tries to send to the servers and it will not be noticed by ecowater.
I am sure that there is an unfixable flaw somewhere in this plan, but maybe it will be an inspiration for an actual solution)

**Comment by barleybobs** [comment]
This should have been fixed in [v4.0.0](https://github.com/barleybobs/homeassistant-ecowater-softener/releases/tag/v4.0.0) 🎉

If this issue reoccurs please do reopen this issue!
